### PR TITLE
Datastore snippets

### PR DESCRIPTION
**Pull request by jskeet**
A few improvements as well, but mostly this is to show how far
we still have to go.

### CODE REVIEW

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 1]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+            CommitResponse response = client.Commit(projectId, Mode.NonTransactional, new[] { entity.ToInsert() });
+            Key insertedKey = response.MutationResults[0].Key;
+            // End snippet
+        }
+
+        [Fact]
+        public void LookupEntity()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: Don't think this is required.

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 2]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+
+            Entity entity = response.Found[0].Entity;
+            // End snippet
+        }
+
+        [Fact]
+        public void UpdateEntity()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: Not required

Comment (jskeet) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: Correct - thanks for spotting. Will see if turning warnings up to max let's me find these in an easy way...

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 3]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+            // End snippet
+        }
+
+        // Snippets ported from https://cloud.google.com/datastore/docs/concepts/queries
+
+        [Fact(Skip = "Requires composite index configuration")]
+        public void CompositeFilterQuery()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: ditto

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 4]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+            // TODO: Results beyond this batch?
+            // End snippet           
+        }
+
+        [Fact]
+        public void KeyQuery()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
+            PartitionId partitionId = _fixture.PartitionId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: And this...

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 5]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+            };
+            // End snippet
+        }
+
+        [Fact]
+        public void AncestorQuery()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
+            PartitionId partitionId = _fixture.PartitionId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: ...

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs [hunk 6]
```diff
@@ -187,5 +267,458 @@ public void Overview()
...
+            };
+            // End snippet
+        }
+
+        [Fact]
+        public void KindlessQuery()
+        {
+            string projectId = _fixture.ProjectId;
+            string namespaceId = _fixture.NamespaceId;
+            PartitionId partitionId = _fixture.PartitionId;
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreClientSnippets.cs]: ..

**Pull request merged**